File: src/test/java/dev/paigewatson/layoutmaster/helpers/RollingStockDALFake.java
package dev.paigewatson.layoutmaster.helpers;

import dev.paigewatson.layoutmaster.data.RollingStockDAL;
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock;

import java.util.List;

public class RollingStockDALFake implements RollingStockDAL
{
    public RollingStock savedEntity;
    public RollingStock deletedEntity;
    private List<RollingStock> returnedFreightCarsList;
    private int deleteAllCalledCount;

    @Override
    public List<RollingStock> getAllRollingStock()
    {
        return returnedFreightCarsList;
    }

    @Override
    public void deleteAll()
    {
        deleteAllCalledCount++;
    }

    @Override
    public void delete(RollingStock rollingStockToDelete)
    {
        deletedEntity = rollingStockToDelete;
    }

    @Override
    public List<RollingStock> getAllOfAARDesignation(AARDesignation aarDesignation)
    {
        return returnedFreightCarsList;
    }

    @Override
    public List<RollingStock> getAllOfRoadName(String roadName)
    {
        return returnedFreightCarsList;
    }

    @Override
    public RollingStock insertRollingStock(RollingStock rollingStockToSave)
    {
        savedEntity = rollingStockToSave;
        return savedEntity;
    }

    public void setReturnedValuesList(List<RollingStock> returnedFreightCarsList)
    {

        this.returnedFreightCarsList = returnedFreightCarsList;
    }

    public int deleteAllCalledCount()
    {
        return deleteAllCalledCount;
    }
}

File: src/test/kotlin/dev/paigewatson/layoutmaster/helpers/RollingStockDALFake.kt
package dev.paigewatson.layoutmaster.helpers

import dev.paigewatson.layoutmaster.data.RollingStockDAL
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock

class RollingStockDALFake : RollingStockDAL {
    var savedEntity: RollingStock? = null
    var deletedEntity: RollingStock? = null
    override var allRollingStock: List<RollingStock> = emptyList()
        private set
    private var deleteAllCalledCount = 0
    override fun deleteAll() {
        deleteAllCalledCount++
    }

    override fun delete(rollingStockToDelete: RollingStock) {
        deletedEntity = rollingStockToDelete
    }

    override fun getAllOfAARDesignation(aarDesignation: AARDesignation): List<RollingStock> {
        return allRollingStock
    }

    override fun getAllOfRoadName(roadName: String): List<RollingStock> {
        return allRollingStock
    }

    override fun insertRollingStock(rollingStockToSave: RollingStock): RollingStock {
        savedEntity = rollingStockToSave
        return savedEntity!!
    }

    fun setReturnedValuesList(returnedFreightCarsList: List<RollingStock>) {
        allRollingStock = returnedFreightCarsList
    }

    fun deleteAllCalledCount(): Int {
        return deleteAllCalledCount
    }
}
